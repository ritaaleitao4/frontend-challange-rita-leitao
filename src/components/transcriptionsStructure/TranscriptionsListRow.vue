<template>
  <div class="list-item">
    <checkbox-item :checkbox-id="item.id" class="checkbox"/>
    <person-svg class="person"/>
    <div class="list-item__container">
      <list-text v-model="item.voice" element="voice"/>
      <list-text v-model="item.text" element="text" type="textArea"/>
    </div>
    <delete-svg class="delete" @click="deleteData(item.id)"/>
  </div>
</template>

<script>
  import { mapActions } from 'vuex'
  import checkboxItem from '@/components/elements/CheckboxItem.vue'
  import listText from '@/components/elements/ListText.vue'
  import deleteSvg from '@/assets/images/delete.svg'
  import personSvg from '@/assets/images/person.svg'

  export default {
    name: 'Transcriptions-list-row',
    components: {
      checkboxItem,
      listText,
      deleteSvg,
      personSvg,
    },
    props: {
      item: {
        required: true,
        type: Object,
      },
    },
    methods: {
      ...mapActions([
        'deleteTranscription',
      ]),
      deleteData(id) {
        this.deleteTranscription(id)
      },
    },
  }
</script>

<style scoped lang="scss">
  .list-item {
    width: 100%;
    display: flex;
    flex-flow: row nowrap;
    padding: $default-padding 53px $default-padding $default-padding;

    &:hover .delete {
      visibility: visible;
      opacity: 1;
    }

    &__container {
      width: 100%;
    }

    svg {
      display: block;

      &.person {
        width: 26px;
        height: 26px;
        margin-right: 8px;
        flex-shrink: 0;
      }

      &.delete {
        position: absolute;
        top: $default-margin;
        right: $default-margin;
        width: 16px;
        height: 20px;
        cursor: pointer;
        visibility: hidden;
        opacity: 0;
        transition: opacity $default-time $default-easing;
      }
    }
  }
</style>
