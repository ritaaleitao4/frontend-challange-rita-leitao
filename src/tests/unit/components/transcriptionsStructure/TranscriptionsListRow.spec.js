import Vue from 'vue'
import Vuex from 'vuex'
import { mount, createLocalVue } from '@vue/test-utils'
import TranscriptionsListRow from '@/components/transcriptionsStructure/TranscriptionsListRow.vue'

const localVue = createLocalVue()
localVue.use(Vuex)
Vue.use(Vuex)

let wrapper
let actions
let store

describe('Transcriptions list add row Test', () => {
  beforeEach(() => {
    actions = {
      deleteTranscription: jest.fn(() => true),
    }

    store = new Vuex.Store({
      actions,
      modules: {
        transcription: {
          namespaced: true,
          actions,
        },
      },
    })

    wrapper = mount(TranscriptionsListRow, {
      stubs: ['list-text', 'person-svg', 'checkbox-item'],
      store,
      localVue,
      propsData: {
        item: {
          id: 1,
          voice: 'Test Voice',
          text: 'Test description',
        },
      },
    })
  })

  it('Snapshot test', () => {
    const Constructor = Vue.extend(TranscriptionsListRow)
    const vm = new Constructor().$mount()
    expect(vm.$el).toMatchSnapshot()
  })

  it('should has a button to delete', () => {
    expect(wrapper.contains('.delete')).toBe(true)
  })

  it('should call action to delete', () => {
    wrapper.find('.delete').trigger('click')
    expect(actions.deleteTranscription).toHaveBeenCalled()
  })
})
