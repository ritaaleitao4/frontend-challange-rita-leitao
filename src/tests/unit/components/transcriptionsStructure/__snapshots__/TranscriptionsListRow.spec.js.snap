// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Transcriptions list add row Test Snapshot test 1`] = `<!---->`;
